x-pack/plugins/observability_solution/infra/server/lib/alerting/inventory_metric_threshold/register_inventory_metric_threshold_rule_type.ts

```diff
@@ -56,13 +54,13 @@ import { O11Y_AAD_FIELDS } from '../../../../common/constants';
 
 const condition = schema.object({
   threshold: schema.arrayOf(schema.number()),
-  comparator: oneOfLiterals(Object.values(Comparator)) as Type<Comparator>,
+  comparator: oneOfLiterals(Object.values(COMPARATORS)) as Type<COMPARATORS>,
```

Comment:
When the inventory rule was introduced, it didn't have "not in between". It was introduced in `v8.10` -I believe by accident due to a types refactoring, and since the 8.10 version, we can't create the rule with "not in between"

> From what I see, it should work fine if someone provides outside in the API

The error we see in the UI is coming from the API. The API will use the validator function provided in the rule registry, so we can't create the rule from the API for the same reason.

Reply:
> The error we see in the UI is coming from the API. The API will use the validator function provided in the rule registry, so we can't create the rule from the API for the same reason.

UI is providing `notBetween,` but in the API validation, it accepts `outside.` When I tried creating a rule directly via API, providing `outside`, the rule was created successfully and alerts were generated as you see below:

![image](https://github.com/elastic/kibana/assets/12370520/1a607e7d-8f96-4705-881a-bacd1deefdd4)

<img src="https://github.com/elastic/kibana/assets/12370520/d345e3a8-e0fb-4f9c-aa8e-231adcc4e2a2" width=500 />

But now the execution is failing:

<img src="https://github.com/elastic/kibana/assets/12370520/7abacf3f-5320-41b4-8aa0-25cebbb27e65" width=500 />